doc/reference/logging/index.rst

```diff
@@ -95,6 +95,11 @@ oldest one are discarded.
 :option:`CONFIG_LOG_MODE_NO_OVERFLOW`: When logger cannot allocate new message
 it is discarded.
 
+:option:`CONFIG_LOG_BLOCK_IN_THREAD`: If enabled and new log message cannot
```

Comment:
I would add some kind of "low water mark" below which thread will block. 
Such mechanism will leave some memory for contexts we cannot block.

Reply:
It's quite difficult to do that. Currently, I used built-in feature of `mem_slab` which blocks when it's empty. What you suggested is more complex. I would prefer to get that in and consider "log water mark" in the future.

Reply:
Ok. Could you please fill the issue suggesting this enhancement?

Reply:
Ping

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/19328